doc/source/rllib/rllib-learner.rst

```diff
@@ -0,0 +1,463 @@
+        :py:meth:`~ray.rllib.core.learner.learner_group.LearnerGroup.set_weights` or
+        :py:meth:`~ray.rllib.core.learner.learner_group.LearnerGroup.get_weights`. 
+        This does not include optimizer states.
+    
+    .. tab-item:: Getting and Setting State for a Learner
+
+        .. testcode::
+            :skipif: True
+
+            # module weights and optimizer states
+            state = learner.get_state()
+            learner.set_state(state)
+
+            # just module state
+            module_state = learner.get_module_state()
+            learner.module.set_module_state(module_state)
+
+        You can set and get the weights of a :py:class:`~ray.rllib.core.learner.learner.Learner` 
+        using :py:meth:`~ray.rllib.core.learner.learner.Learner.set_state` 
+        and :py:meth:`~ray.rllib.core.learner.learner.Learner.get_state` .
+        For setting or getting only RLModule weights (without optimizer states), use 
+        :py:meth:`~ray.rllib.core.learner.learner.Learner.set_module_state` 
+        or :py:meth:`~ray.rllib.core.learner.learner.Learner.get_module_state` API.
+
+
+.. testcode::
+	:hide:
+    :skipif: True
+
+	import shutil
+	import tempfile
+
+	LEARNER_CKPT_DIR = str(tempfile.TemporaryDirectory())
+	LEARNER_GROUP_CKPT_DIR = str(tempfile.TemporaryDirectory())
+
+
+Checkpointing
+-------------
+
+.. tab-set::
+    
+    .. tab-item:: Checkpointing a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            learner_group.save_state(LEARNER_GROUP_CKPT_DIR)
+            learner_group.load_state(LEARNER_GROUP_CKPT_DIR)
+        
+        Checkpoint the state of all learners in the :py:class:`~ray.rllib.core.learner.learner_group.LearnerGroup` via :py:meth:`~ray.rllib.core.learner.learner_group.LearnerGroup.save_state` and
+        :py:meth:`~ray.rllib.core.learner.learner_group.LearnerGroup.load_state`. This includes all states including neural network weights and any
+        optimizer states. Note that since the state of all of the :py:class:`~ray.rllib.core.learner.learner.Learner` instances is identical,
+        only the states from the first :py:class:`~ray.rllib.core.learner.learner.Learner` need to be saved.
+
+    .. tab-item:: Checkpointing a Learner
+
+        .. testcode::
+            :skipif: True
+
+            learner.save_state(LEARNER_CKPT_DIR)
+            learner.load_state(LEARNER_CKPT_DIR)
+
+        Checkpoint the state of a :py:class:`~ray.rllib.core.learner.learner.Learner` 
+        via :py:meth:`~ray.rllib.core.learner.learner.Learner.save_state` and 
+        :py:meth:`~ray.rllib.core.learner.learner.Learner.load_state`. This 
+        includes all states including neural network weights and any optimizer states.
+
+
+Implementation
+==============
+:py:class:`~ray.rllib.core.learner.learner.Learner` has many APIs for flexible implementation, however the core ones that you need to implement are:
```

Comment:
The Learner class has a lot of abstract methods (13 or so).
I think we should clarify here that if you would want to only override the ones below, you'll have to subclass a framework specific implementation.

Reply:
I think what I was going for here is to describe the minimum functions that people need to implement or override for their custom learners, whether that be them overriding them or implementing a brand new one.

Reply:
Imo this should read:

Writing or customizing Learners
===================

> If you want to write your own custom :py:class:`~ray.rllib.core.learner.learner.Learner`, you will want to subclass a framework-specific Learner. The following is a list of the most important methods that you'll want to implement. Similarely, these methods are likely what you'll want to override if you want to modify an existing Learner like TorchPPOLearner <API ref link> :

(...)